src/plugin.rs

```diff
@@ -110,30 +203,61 @@ impl MigrationSource<'static> for MigrationList {
 #[command]
 async fn load(
     db_instances: State<'_, DbInstances>,
+    locations: State<'_, Locations>,
     migrations: State<'_, Migrations>,
     db: String,
-) -> Result<()> {
-    if !Db::database_exists(&db).await.unwrap_or(false) {
-        Db::create_database(&db).await?;
+) -> Result<String> {
+    let conn = Connection::from(db)
+        .add_app_directory(&locations.app)
+        .to_string();
+
+    let db_exists = Db::database_exists(&conn).await.unwrap_or(false);
+    let app_dir_exists = Path::new(&locations.app).is_dir();
+
+    if !app_dir_exists {
```

Comment:
We need to ensure the app directory does indeed exist and create it if it doesn't

Reply:
ok, that sounds better

Reply:
yep. It creates all dirs recursively (like mkdir -p i think) and doesn't throw an error if one of them already exists.

Reply:
this has been done